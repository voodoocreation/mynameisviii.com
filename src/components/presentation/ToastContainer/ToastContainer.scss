@import "../../../scss/variables";

.ToastContainer {
  display: flex;
  flex-direction: column;
  justify-content: left;
  position: fixed;
  right: $spacing;
  bottom: $spacing;
  left: $spacing;
  z-index: ($modal-index + 1);
}
