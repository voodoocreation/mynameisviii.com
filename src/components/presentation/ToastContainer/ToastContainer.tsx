import * as React from "react";

import NewVersionToast from "../../connected/NewVersionToast/NewVersionToast";
import OnlineStatusToast from "../../connected/OnlineStatusToast/OnlineStatusToast";

import "./ToastContainer.scss";

const ToastContainer: React.FC = () => (
  <section className="ToastContainer">
    <OnlineStatusToast />
    <NewVersionToast />
  </section>
);

export default ToastContainer;
