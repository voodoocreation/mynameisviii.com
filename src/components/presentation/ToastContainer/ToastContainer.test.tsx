import { render } from "enzyme";
import * as React from "react";

import ToastContainer from "./ToastContainer";

const setup = (fn: any, fromTestProps?: any) => {
  const props = {
    className: "TestToastContainer",
    ...fromTestProps
  };

  return {
    actual: fn(<ToastContainer {...props} />),
    props
  };
};

describe("[presentation] <ToastContainer />", () => {
  it("renders correctly with children", () => {
    const { actual } = setup(render, { children: <div>Test</div> });

    expect(actual.find("div")).toHaveLength(1);
    expect(actual).toMatchSnapshot();
  });

  it("renders null with no children", () => {
    const { actual } = setup(render);

    expect(actual.find(".ToastContainer")).toHaveLength(0);
    expect(actual).toMatchSnapshot();
  });
});
