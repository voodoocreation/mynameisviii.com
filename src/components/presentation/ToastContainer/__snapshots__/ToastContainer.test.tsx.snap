// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[presentation] <ToastContainer /> renders correctly with children 1`] = `
<div
  class="ToastContainer TestToastContainer"
>
  <div>
    Test
  </div>
</div>
`;

exports[`[presentation] <ToastContainer /> renders null with no children 1`] = `null`;
